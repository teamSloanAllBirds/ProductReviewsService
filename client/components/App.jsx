import React from 'react';
import ReactDOM from 'react-dom';
import $ from 'jquery';
import axios from 'axios';

// import Review from './Review.jsx';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      reviews: [],
      visible: 3
    };

    this.handleFetchReviews = this.handleFetchReviews.bind(this);
    this.loadMore = this.loadMore.bind(this);
  }

  componentWillMount() {
    this.handleFetchReviews();
  }

  handleFetchReviews() {
    axios.get('/api/reviews')
      .then(({data}) => {
        console.log('CLIENT GET SUCCESS (reviews)', data);
        this.setState({reviews: data});
      });
  }

  loadMore() {
    this.setState((prev) => {
      return {visible: prev.visible + 3};
    });
  }

  render() {
    return (
      <div>
        <h1>Feedback from the Flock</h1>
        <div className='app'>
          {/* <Review
            reviews={this.state.reviews}
            onFetcheReview={this.handleFetchReviews}
          /> */}
          <div className='test'>
            {this.state.reviews.slice(0, this.state.visible).map((review, index) => {
              return (
                <div key={review.id}>
                  <h2>Count: {index + 1}</h2>
                  <div>{review.customerName}</div>
                  <div>{review.createdAt}</div>
                  <div>{review.rating}</div>
                  <div>{review.commentTitle}</div>
                  <div>{review.comment}</div>
                </div>
              );
            })}
            {this.state.visible < this.state.reviews.length &&
              <button type='button' onClick={this.loadMore}>
                Load More
              </button>
            }
          </div>
        </div>
      </div>
    );
  }
}

export default App;



