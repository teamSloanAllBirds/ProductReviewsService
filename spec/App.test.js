import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJSON from 'enzyme-to-json';

import App from '../client/components/App.jsx';


describe('App functionality', () => {
  it('should render without throwing an error', function() {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });

  it('should be selectable by className "reviews-container"', () => {
    // eslint-disable-next-line func-style
    function callback(data) {
      const component = shallow(<App />);
      expect(component.is('.reviews-container')).toBe(true);
      done();
    }
  });

  it('should render and mount properly', () => {
    // eslint-disable-next-line func-style
    function callback(data) {
      const component = mount(<App />);
      expect(component).toMatchSnapshot();
      done();
    }
  });

  it('should mount in a full DOM', function() {
    // eslint-disable-next-line func-style
    function callback(data) {
      const component = mount(<App />);
      expect(component.find('.app').length).toBe(1);
      done();
    }
  });
});