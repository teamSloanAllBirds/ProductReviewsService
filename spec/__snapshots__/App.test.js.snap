// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App functionality should render without throwing an error 1`] = `ShallowWrapper {}`;
